src/ImageSharp/Image/ImageBase{TColor}.cs

```diff
@@ -162,13 +162,15 @@ public virtual PixelAccessor<TColor> Lock()
         internal void SwapPixelsBuffers(PixelAccessor<TColor> pixelSource)
         {
             Guard.NotNull(pixelSource, nameof(pixelSource));
-            Guard.IsTrue(pixelSource.PooledMemory, nameof(pixelSource.PooledMemory), "pixelSource must be using pooled memory");
 
+            // TODO: Was this check really useful? If yes, we can define a bool PixelAccessor<TColor>.IsBoundToImage to re-introduce it:
```

Comment:
Thought as much thanks! @antonfirsov I've we are still able to lock an array then we should reintroduce the check in the manner you suggested in your comments.